lib/galaxy/jobs/actions/post.py

```diff
@@ -242,6 +252,16 @@ def execute(cls, app, sa_session, action, job, replacement_dict):
             if action.output_name == '' or dataset_assoc.name == action.output_name:
                 dataset_assoc.dataset.deleted = True
 
+        for dataset_collection_assoc in job.output_dataset_collection_instances:
```

Comment:
Hmm, I like the DeleteDatasetAction option, but as per the first comment it says this is disabled ?!
I think it's not actually disabled, but I have experienced situations where intermediate data got deleted too early ... I'll see if I can write a testcase for that.

Reply:
Alright, I get it, it is disabled because it's not listed in the `ActionBox`

Reply:
I confused this with `Output cleanup`, I think, so that's unrelated.